### PR TITLE
Diligen api client

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

close [FLA-758](https://justice.gov.bc.ca/jira/browse/FLA-758)

This PR includes the following proposed change(s):

- Adding Diligen health indicators on the api health checks

image

The diligen client is autoconfigured via application properties:

```yml
jag:
  efiling:
    diligen:
      health:
        enabled: ${DILIGEN_HEALTH_ENABLED:true}
      basePath: ${DILIGEN_BASE_PATH}
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-diligen-client-starter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                <configuration>
+                    <source>${java.version}</source>
+                    <target>${java.version}</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+
+</project>
```
Comment (alexjoybc) [inline src/backend/libs/efiling-diligen-client-starter/pom.xml:86]:
```suggestion
</project>

```

File: src/backend/libs/efiling-diligen-client-starter/src/main/resources/META-INF/spring.factories [hunk 2]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+    ca.bc.gov.open.jag.efilingdiligenclientstarter.AutoConfiguration
```
Comment (TayGov) [inline src/backend/libs/efiling-diligen-client-starter/src/main/resources/META-INF/spring.factories:2]: newline

Comment (TayGov) [reply src/backend/libs/efiling-diligen-client-starter/src/main/resources/META-INF/spring.factories:2]:
```suggestion
    ca.bc.gov.open.jag.efilingdiligenclientstarter.AutoConfiguration

```

File: src/backend/efiling-reviewer-api/Efiling Review Api.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+						"url": {
+							"raw": ""
+						}
+					},
+					"response": []
+				}
+			]
+		}
+	]
+}
```
[the remark this answers is not in the document]
Comment (TayGov) [reply src/backend/efiling-reviewer-api/Efiling Review Api.postman_collection.json:25]:
```suggestion
}

```

File: src/backend/pom.xml [hunk 4]
```diff
@@ -29,6 +29,8 @@
 				<module>libs/efiling-cso-starter</module>
 				<module>libs/efiling-bom</module>
 				<module>libs/efiling-demo-starter</module>
+				<module>libs/efiling-diligen-client</module>
+				<module>libs/efiling-diligen-client-starter</module>
```
Comment (TayGov) [inline src/backend/pom.xml:33]: Might want to add these to the efiling-reviewer profile. It won't let me comment there to suggest code....

**Review by TayGov — CHANGES_REQUESTED** [review]
Just a couple minor things

File: src/backend/efiling-reviewer-api/src/main/resources/application.yml [hunk 5]
```diff
@@ -1,2 +1,12 @@
...
+management:
+  endpoint:
+    health:
+      show-details: always
+jag:
+  efiling:
+    diligen:
+      health:
+        enabled: ${DILIGEN_HEALTH_ENABLED:true}
+      basePath: ${DILIGEN_BASE_PATH}
```
[the remark this answers is not in the document]
Comment (TayGov) [reply src/backend/efiling-reviewer-api/src/main/resources/application.yml:12]: Neve mind I see it is intended. Just triggering my ocd

**Review by TayGov — APPROVED** [review]

**Pull request merged**